homeassistant/components/tplink/light.py

```diff
@@ -177,6 +181,13 @@ def is_on(self):
 
     def update(self):
         """Update the TP-Link Bulb's state."""
+        if self._supported_features is None:
+            self.do_update()
+        else:
+            self.hass.async_run_job(self.do_update)
```

Comment:
You cannot call async functions inside sync functions. 

We should also not call this method non-blocking, because Home Assistant writes the state to the state machine once the update is done.

Reply:
The latest change should address this.